src/group_theory/order_of_element.lean

```diff
@@ -400,6 +400,24 @@ end
 
 end group
 
+section comm_group
+
+variables [comm_group G]
+
+/-- Elements of finite order are closed under multiplication. -/
+@[to_additive "Elements of finite order are closed under addition."]
+lemma is_of_fin_order.mul (hx : is_of_fin_order x) (hy : is_of_fin_order y) :
```

Comment:
I suggest you provide a version with `[group G]` and `commute x y` assumptions, and derive this from that.

Reply:
Nice! I like that, hadn't seen `commute` before but I think done now. Thanks for the suggestion.

Reply:
I just found that we actually already have [commute.order_of_mul_dvd_lcm](https://leanprover-community.github.io/mathlib_docs/group_theory/order_of_element.html#commute.order_of_mul_dvd_lcm) and [commute.order_of_mul_dvd_mul_order_of](https://leanprover-community.github.io/mathlib_docs/group_theory/order_of_element.html#commute.order_of_mul_dvd_mul_order_of), and finite-orderness can be derived from either of them

Reply:
Does my latest attempt seem ok there, I tried to use `order_of_mul_dvd_mul_order_of`?

Reply:
Looks nice, thanks! One more thing: I think the assumption here can be `[comm_monoid G]` here, and it would automatically apply to `[comm_group G]` as well.

Reply:
Aha, indeed, you're correct again! Done.